craft_providers/multipass/multipass.py

```diff
@@ -0,0 +1,382 @@
+
+        quoted_final_cmd = shlex.join(final_cmd)
+        logger.debug("Executing on host: %s", quoted_final_cmd)
+
+        return runner(final_cmd, **kwargs)  # pylint: disable=subprocess-run-check
+
+    def info(self, *, instance_name: str) -> Optional[Dict[str, Any]]:
+        """Get information/state for instance.
+
+        :returns: Parsed json data from info command.
+
+        :raises MultipassError: On error.
+        """
+        command = ["info", instance_name, "--format", "json"]
+
+        try:
+            proc = self._run(command, capture_output=True, check=True, text=True)
+        except subprocess.CalledProcessError as error:
+            if "does not exist" in error.stderr:
+                return None
+
+            raise MultipassError(
+                brief=f"Failed to query info for VM {instance_name!r}.",
+                details=errors.details_from_called_process_error(error),
+            ) from error
+
+        return json.loads(proc.stdout)
+
+    def launch(
+        self,
+        *,
+        instance_name: str,
+        image: str,
+        cpus: str = None,
+        mem: str = None,
+        disk: str = None,
+    ) -> None:
+        """Launch multipass VM.
+
+        :param instance_name: The name the launched instance will have.
+        :param image: Name of image to create the instance with.
+        :param cpus: Amount of virtual CPUs to assign to the launched instance.
+        :param mem: Amount of RAM to assign to the launched instance.
+        :param disk: Amount of disk space the launched instance will have.
+
+        :raises MultipassError: on error.
+        """
+        command = ["launch", image, "--name", instance_name]
+        if cpus is not None:
+            command.extend(["--cpus", cpus])
+        if mem is not None:
+            command.extend(["--mem", mem])
+        if disk is not None:
+            command.extend(["--disk", disk])
+
+        try:
+            self._run(command, capture_output=True, check=True)
+        except subprocess.CalledProcessError as error:
+            raise MultipassError(
+                brief=f"Failed to launch VM {instance_name!r}.",
+                details=errors.details_from_called_process_error(error),
+            ) from error
+
+    def list(self) -> List[str]:
+        """List names of VMs.
+
+        :returns: Data from stdout if instance exists, else None.
+
+        :raises MultipassError: On error.
+        """
+        command = ["list", "--format", "json"]
+
+        try:
+            proc = self._run(command, capture_output=True, check=True, text=True)
+        except subprocess.CalledProcessError as error:
+            raise MultipassError(
+                brief="Failed to query list of VMs.",
+                details=errors.details_from_called_process_error(error),
+            ) from error
+
+        data_list = json.loads(proc.stdout).get("list", dict())
```

Comment:
The default if `"list"` is not present should be a list, right? Not a dict.

Reply:
Yes indeed, though never quite happens.  Just a sane fallback.
```
$ multipass list --format=json
{
    "list": [
    ]
}
```

Reply:
fixed